### PR TITLE
2913: Update domain request screenreader outputs [ES]

### LINKED ISSUE #2913: Bundle: Request form: Screen reader output
### Story
As an assistive technology enabled uder, I want the screen reader output on the "Other employees from your organization" and "Add'l details" sections of the domain request form to be properly formatted and accessible, so that I can navigate and complete the form efficiently and without confusion.

### Acceptance Criteria
- [x] #2914
- [x] #2915
- [x] #2918
- [x] On the Other employees from your organization page, move the _Required fields are marked with an asterisk (*)._ below _We typically don’t reach out to these employees, but if contact is necessary, our practice is to coordinate with you first._
- [x] #2921
- [x] #2922
- [x] #2923 

### Additional Context

Reminder: no unit tests are required for screen reader output
How to view screen reader output using ANDI
- [Download ANDI](https://www.ssa.gov/accessibility/andi/help/install.html)
- Open ANDI
- Navigate to focusable elements
- Use the ⏪  ⏩ buttons to navigate through the page or hover over the element you want to inspect
- Use this handy [ANDI key](https://docs.google.com/document/d/1WTFBFvycmi8DjjxAVHjRjTIuq-QYkwCSNsN1XVrcs54/edit) to understand how the ANDI output colors map to code `attributes` and `<elements>`.

Screenshot 2024-10-08 at 9 00 09 AM

NOTE: No unit tests are required for screen reader output.
### Issue Links
- Caused by #2866 
- Relates to #2924 

### Prioritization Comments
**WCAG Conformance Level:** 2.1 AA
**WCAG Standard that applies:** 
1.3.1 - Info and Relationships (Critical)
3.3.2 - Labels or Instructions (Critical)
4.1.2 - Name, Role, Value (Critical)
**Critical Failure if Not Met?:**  Yes
**Prioritization:** Blocker
**Reason:** This story addresses critical accessibility issues that affect users relying on assistive technologies. In this context, the reader output is not properly formatted and does not provide effective form navigation and completion. Mis-formatted outputs can lead to confusion, resulting in a negative user experience and potential failure to submit necessary information.

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #2913 

## Changes

- Add screenreader output changes included in ticket. #2914 was resolved in a separate PR before it was decided we'd put all our subissues in a single PR 

## Setup

With org model feature turned off, go to a domain request form and verify that all screenreader outputs match the subissues.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests - NA
- [x] Update documentation in READMEs and/or onboarding guide - NA

#### Ensured code standards are met (Original Developer)

- [x] If any updated dependencies on Pipfile, also update dependencies in requirements.txt. - NA
- [x] Interactions with external systems are wrapped in try/except
- [x] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [x] Tag @dotgov-designers in this PR's Reviewers for design review. If code is not user-facing, delete design reviewer checklist
- [x] Verify new pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Verified code meets all checks above. Address any checks that are not satisfied
- [ ] Reviewed this code and left comments. Indicate if comments must be addressed before code is merged
- [ ] Checked that all code is adequately covered by tests
- [ ] Verify migrations are valid and do not conflict with existing migrations

#### Validated user-facing changes as a developer
**Note:** Multiple code reviewers can share the checklists above, a second reviewer should not make a duplicate checklist. All checks should be checked before approving, even those labeled N/A. 

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] (Rarely needed) Tested as both an analyst and applicant user

#### Verified that the changes match the design intention

- [x] Checked that the design translated visually
- [ ] Checked behavior. Comment any found issues or broken flows.
- [x] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links

#### Validated user-facing changes as a designer

- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [x] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [x] Safari
- [ ] (Rarely needed) Tested as both an analyst and applicant user

### References
- [Code review best practices](../docs/dev-practices/code_review.md)

### CODE REVIEW

File: src/registrar/assets/sass/_theme/_typography.scss [hunk 1]
```diff
@@ -27,6 +27,10 @@ h2 {
 .usa-form,
 .usa-form fieldset {
   font-size: 1rem;
+  legend em {
```
Comment (erinysong) [inline src/registrar/assets/sass/_theme/_typography.scss:30]: Now that our field headings and subheadings are included as legend items in the screenreader fix, we're setting font-size to 1rem to preserve their font size 16px.

File: src/registrar/assets/src/sass/_theme/_base.scss [hunk 2]
```diff
@@ -149,6 +149,11 @@ footer {
   color: color('primary');
 }
 
+.usa-radio {
+  margin-top: 1rem;
+  font-size: 1.06rem;
```
Comment (erinysong) [inline src/registrar/assets/src/sass/_theme/_base.scss:154]: Added to preserve styling of our radio buttons (bigger text, margins)

File: src/registrar/assets/src/sass/_theme/_typography.scss [hunk 3]
```diff
@@ -34,6 +34,9 @@ h2 {
 .usa-form,
 .usa-form fieldset {
   font-size: 1rem;
+  .usa-legend {
```
Comment (erinysong) [inline src/registrar/assets/src/sass/_theme/_typography.scss:37]: Now that our field headings and subheadings are moved to the legends in the screenreader fix, we're updating the font size of our legends to maintain our font size of 16px (originally 16.96px from usa-legend font size default being 1.06rem)

File: src/registrar/templates/django/forms/label.html [hunk 4]
```diff
@@ -2,15 +2,25 @@
   class="{% if label_classes %} {{ label_classes }}{% endif %}{% if label_tag == 'legend' %} {{ legend_classes }}{% endif %}"
   {% if not field.use_fieldset %}for="{{ widget.attrs.id }}"{% endif %}
 >
-  {% if span_for_text %}
-    <span>{{ field.label }}</span>
+  {% if legend_heading %}
```
Comment (erinysong) [inline src/registrar/templates/django/forms/label.html:5]: Integrates legends into our labels so that their contents are tied to screenreader output

**Comment by erinysong** [comment]
Question - do we want expand these screenreader outputs to the portfolio forms as well? I can change them wherever applicable but feel free to let me know if we want this only scoped to our non org model domain request form

File: src/registrar/templates/domain_request_other_contacts.html [hunk 5]
```diff
@@ -9,28 +9,22 @@
...
 
 {% block form_fields %}
-    <fieldset class="usa-fieldset margin-top-2">
-      <legend>
-        <h2>Are there other employees who can help verify your request?</h2>
-      </legend>
-
-      {% with add_class="usa-radio__input--tile" add_legend_class="usa-sr-only" %}
+    <div class="margin-top-2">
+      {% with add_class="usa-radio__input--tile" add_legend_heading="Are there other employees who can help verify your request?" %}
```
Comment (erinysong) [inline src/registrar/templates/domain_request_other_contacts.html:21]: #2915

File: src/registrar/forms/domain_request_wizard.py [hunk 6]
```diff
@@ -736,7 +736,14 @@ class NoOtherContactsForm(BaseDeletableRegistrarForm):
         required=True,
         # label has to end in a space to get the label_suffix to show
         label=("No other employees rationale"),
-        widget=forms.Textarea(),
+        widget=forms.Textarea(
+            attrs={
+                "aria-label": "You don’t need to provide names of other employees now, \
```
Comment (erinysong) [inline src/registrar/forms/domain_request_wizard.py:741]: #2918 and #2915

File: src/registrar/templates/domain_request_additional_details.html [hunk 7]
```diff
@@ -9,15 +9,9 @@
...
-      <legend>
-        <h2>Are you working with a CISA regional representative on your domain request?</h2>
-        <p>.gov is managed by the Cybersecurity and Infrastructure Security Agency. CISA has <a href="https://www.cisa.gov/about/regions" target="_blank">10 regions</a> that some organizations choose to work with. Regional representatives use titles like protective security advisors, cyber security advisors, or election security advisors.</p>
-      </legend>
-
+    <fieldset class="usa-fieldset">
       <!-- Toggle -->
-      <em>Select one. <abbr class="usa-hint usa-hint--required" title="required">*</abbr></em>
-      {% with add_class="usa-radio__input--tile" add_legend_class="usa-sr-only" %}
+      {% with add_class="usa-radio__input--tile" add_legend_class="margin-top-0" add_legend_heading="Are you working with a CISA regional representative on your domain request?" %}
```
Comment (erinysong) [inline src/registrar/templates/domain_request_additional_details.html:14]: #2921

File: src/registrar/templates/django/forms/label.html [hunk 8]
```diff
@@ -2,15 +2,25 @@
...
+    <h2 class="{{ legend_classes }}">{{ legend_heading }} </h2>
+    {% if widget.attrs.id == 'id_additional_details-has_cisa_representative' %}
+      <p>.gov is managed by the Cybersecurity and Infrastructure Security Agency. CISA has <a href="https://www.cisa.gov/about/regions" target="_blank">10 regions</a> that some organizations choose to work with. Regional representatives use titles like protective security advisors, cyber security advisors, or election security advisors.</p>
```
Comment (erinysong) [inline src/registrar/templates/django/forms/label.html:8]: #2921 #2922

File: src/registrar/templates/domain_request_additional_details.html [hunk 9]
```diff
@@ -31,21 +25,16 @@ <h2>Are you working with a CISA regional representative on your domain request?<
...
 
     <fieldset class="usa-fieldset margin-top-2">
-        <legend>
-            <h2>Is there anything else you’d like us to know about your domain request?</h2> 
-        </legend>
-  
         <!-- Toggle -->
-        <em>Select one. <abbr class="usa-hint usa-hint--required" title="required">*</abbr></em>
-        {% with add_class="usa-radio__input--tile" add_legend_class="usa-sr-only" %}
+        {% with add_class="usa-radio__input--tile" add_legend_heading="Is there anything else you’d like us to know about your domain request?" %}
```
Comment (erinysong) [inline src/registrar/templates/domain_request_additional_details.html:29]: #2923

**Review by SamiyahKey — COMMENTED** [review]
Approved, thanks Erin!

File: src/registrar/forms/domain_request_wizard.py (hunk 6 above)
Comment (asaki222) [reply src/registrar/forms/domain_request_wizard.py:741]: @erinysong I don't see the "No other employees" text in the ANDI Output mentioned in #2918 
Screenshot 2024-12-19 at 9 31 38 AM

Do I need to focus on the text area differently?

**Review by asaki222 — APPROVED** [review]
LGTM!

**Pull request merged**